components/script/dom/stereopannernode.rs

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::dom::audioparam::AudioParam;
+use crate::dom::audioscheduledsourcenode::AudioScheduledSourceNode;
+use crate::dom::baseaudiocontext::BaseAudioContext;
+use crate::dom::bindings::codegen::Bindings::AudioNodeBinding::{
+    ChannelCountMode, ChannelInterpretation,
+};
+use crate::dom::bindings::codegen::Bindings::AudioParamBinding::AutomationRate;
+use crate::dom::bindings::codegen::Bindings::StereoPannerNodeBinding::StereoPannerNodeMethods;
+use crate::dom::bindings::codegen::Bindings::StereoPannerNodeBinding::{
+    self, StereoPannerOptions,
+};
+use crate::dom::bindings::error::Fallible;
+use crate::dom::bindings::reflector::reflect_dom_object;
+use crate::dom::bindings::root::{Dom, DomRoot};
+use crate::dom::window::Window;
+use dom_struct::dom_struct;
+use servo_media::audio::stereo_panner_node::StereoPannerOptions as ServoMediaStereoPannerOptions;
+use servo_media::audio::node::AudioNodeInit;
+use servo_media::audio::param::ParamType;
+use std::f32;
+
+#[dom_struct]
+pub struct StereoPannerNode {
+    source_node: AudioScheduledSourceNode,
+    offset: Dom<AudioParam>,
+}
+
+impl StereoPannerNode {
+    #[allow(unrooted_must_root)]
+    pub fn new_inherited(
+        window: &Window,
+        context: &BaseAudioContext,
+        options: &StereoPannerOptions,
+    ) -> Fallible<StereoPannerNode> {
+        let node_options =
+            options
+                .parent
+                .unwrap_or(2, ChannelCountMode::Max, ChannelInterpretation::Speakers);
+        let source_node = AudioScheduledSourceNode::new_inherited(
```

Comment:
You need to validate based on constraints here, see https://webaudio.github.io/web-audio-api/#audionode-channelcount-constraints , https://webaudio.github.io/web-audio-api/#dom-audionode-channelcountmode : if the channel count mode is Max or the channel count is greater than two, return a NotSupported Error

You'll also need to add similar constraints on SetChannelCount and SetChannelCountMode on audionode.rs